src/jit/hwintrinsicxarch.cpp

```diff
@@ -213,36 +252,175 @@ GenTree* Compiler::impX86HWIntrinsic(NamedIntrinsic intrinsic, CORINFO_METHOD_HA
+    {
+        case TYP_FLOAT:
+            if (simdType == TYP_SIMD16)
+            {
+                return Vector128FloatHandle;
+            }
+            else if (simdType == TYP_SIMD32)
+            {
+                return Vector256FloatHandle;
+            }
+            break;
+        case TYP_DOUBLE:
+            if (simdType == TYP_SIMD16)
+            {
+                return Vector128DoubleHandle;
+            }
+            else if (simdType == TYP_SIMD32)
+            {
+                return Vector256DoubleHandle;
+            }
+            break;
+        case TYP_INT:
+            if (simdType == TYP_SIMD16)
+            {
+                return Vector128IntHandle;
+            }
+            else if (simdType == TYP_SIMD32)
+            {
+                return Vector256IntHandle;
+            }
+            break;
+        case TYP_CHAR:
+        case TYP_USHORT:
+            if (simdType == TYP_SIMD16)
+            {
+                return Vector128UShortHandle;
+            }
+            else if (simdType == TYP_SIMD32)
+            {
+                return Vector256UShortHandle;
+            }
+            break;
+        case TYP_UBYTE:
+            if (simdType == TYP_SIMD16)
+            {
+                return Vector128UByteHandle;
+            }
+            else if (simdType == TYP_SIMD32)
+            {
+                return Vector256UByteHandle;
+            }
+            break;
+        case TYP_SHORT:
+            if (simdType == TYP_SIMD16)
+            {
+                return Vector128ShortHandle;
+            }
+            else if (simdType == TYP_SIMD32)
+            {
+                return Vector256ShortHandle;
+            }
+            break;
+        case TYP_BYTE:
+            if (simdType == TYP_SIMD16)
+            {
+                return Vector128ByteHandle;
+            }
+            else if (simdType == TYP_SIMD32)
+            {
+                return Vector256ByteHandle;
+            }
+            break;
+        case TYP_LONG:
+            if (simdType == TYP_SIMD16)
+            {
+                return Vector128LongHandle;
+            }
+            else if (simdType == TYP_SIMD32)
+            {
+                return Vector256LongHandle;
+            }
+            break;
+        case TYP_UINT:
+            if (simdType == TYP_SIMD16)
+            {
+                return Vector128UIntHandle;
+            }
+            else if (simdType == TYP_SIMD32)
+            {
+                return Vector256UIntHandle;
+            }
+            break;
+        case TYP_ULONG:
+            if (simdType == TYP_SIMD16)
+            {
+                return Vector128ULongHandle;
+            }
+            else if (simdType == TYP_SIMD32)
+            {
+                return Vector256ULongHandle;
+            }
+            break;
+        default:
+            assert(!"Didn't find a class handle for simdType");
+    }
+    return NO_CLASS_HANDLE;
+}
+
```

Comment:
I think this might be more readable if the `if`s happened first and the switches were per-type.

Also it seems a bit odd that we just return these cached handle values without ensuring that the are non null. I suppose there is an implicit guarantee that the type we look up here will have been previously looked up by the code that does set the cache, but it might be nice to assert that the cached results are not null.

Reply:
Will make these two changes.